algorithm.go

```diff
@@ -0,0 +1,151 @@
+package cose
+
+import (
+	"crypto"
+	"hash"
+	"strconv"
+)
+
+// Algorithms supported by this library.
+const (
+	AlgorithmPS256   Algorithm = -37 // RSASSA-PSS w/ SHA-256 by RFC 8230
+	AlgorithmPS384             = -38 // RSASSA-PSS w/ SHA-384 by RFC 8230

```

Comment:
Types are only propagated when using `iota`. In this const block only `AlgorithmPS256` is an Algorithm, others are untyped constants.